### PR TITLE
multi_heap: keep number of heaps as an unsigned value

### LINKED ISSUE #67969: [Coverity CID: 240685] Negative array index read in lib/heap/multi_heap.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/69522b8694e11a0597935169c7d5c0ece0309496/lib/heap/multi_heap.c

Category: Memory - illegal accesses
Function: `sys_multi_heap_add_heap`
Component: Other
CID: [240685](https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=240685)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/69522b8694e11a0597935169c7d5c0ece0309496/lib/heap/multi_heap.c#L38

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/#/project-view/29271/12996?selectedIssue=240685

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_NEGATIVE_RETURNS). ([CWE-129](http://cwe.mitre.org/data/definitions/129.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### PR DESCRIPTION
**Pull request by npitre**
Although very unlikely for this value to ever be negative in practice,
this would make Coverity happy.

Fixes #67969

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Review by peter-mitsis — APPROVED** [review]
I have a hard time remembering when 'int' vs 'unsigned int' comparisons have the potential to get flagged by either the compiler or static analysis tools. That disclaimer out of the way ... would it be worthwhile to also update the relevant loop counters / heap indexes in lib/os/multi_heap.c from 'int' to 'unsigned int' to ensure that static analysis tools remain happy?

**Comment by npitre** [comment]
> would it be worthwhile to also update the relevant loop counters / heap 
> indexes in lib/os/multi_heap.c from 'int' to 'unsigned int' to ensure 
> that static analysis tools remain happy?

Well... All the static analyser could prove is that `mheap->nheaps` was 
initially constrained between `INT_MIN` and `ARRAY_SIZE(mheap->heaps) - 1`.
Now it is between 0 and `ARRAY_SIZE(mheap->heaps) - 1`. So any decent 
static analyser shouldn't care any longer whether or not the loop index 
is signed or not given it is initialized to 0, always increased, and 
again upper bounded to `ARRAY_SIZE(mheap->heaps) - 1`.

In other words, I don't think it would be necessary. It is a very common 
C idiom to use plain ints with loops so static analisers better be smart 
enough recognizing those cases where it is OK to do so.

**Review by nashif — APPROVED** [review]
